### ISSUE TITLE
Create a "roadmap-ish" GitHub project

### ISSUE THREAD

**Issue by negz** [crossplane/crossplane#2526]
### What problem are you facing?

Crossplane is a community driven project. Historically we haven't been able to reliably forecast how much time community members can commit to the project, and thus have not maintained a roadmap with specific time commitments. Currently we maintain a GitHub project board for each Crossplane release, with one (minor version) release every eight weeks. These release scoped project boards primarily focus on "core Crossplane" - i.e. they don't track work that is scoped to a specific provider such as building out a new managed resource.

Relying only on release scoped projects makes it difficult for Crossplane community members to get a sense of the things the project has planned beyond the next release. It also makes it difficult for Crossplane community members to get a reliable sense of what will be included in the next release, because each release project tends to include:

* Issues tracking work that is happening _during_ the development cycle of a particular release, but that is not intended to _actually ship_ as part of the release. Design documents for larger features are a common example of this practice.
* A long, roughly prioritised, list of "proposed" work. In theory this is work that is proposed for each release, but in practice much of it is dragged from release project to release project as we wait for someone to have the bandwidth to tackle it.

### How could Crossplane help solve your problem?

Crossplane could maintain a "roadmap" project board. Note that despite the name this board would _not_ include time commitments. Instead it would use three columns:

* __Proposed__: Work that is proposed for inclusion in an unspecified future release.
* __In Design__: Work that is still in the design and discussion phase, without a commitment for the proposed functionality to ship with any particular release.
* __In Progress__: Work that is in progress but that not associated with a particular release of Crossplane. A prime example of this is the work being done on https://github.com/crossplane-contrib/terrajet to automate the process of building new providers.

Notably the "Proposed" column would be approximately ordered by priority, with items closer to the top of the column being those of higher priority.

**Comment by muvaf** [comment]
I agree that a running board would be more suitable. Would labeling the ones we want to include in a specific release as such make sense here? Possibly the ones under **In Progress**, like `release-1.5`.
